biz.aQute.bndlib/src/aQute/bnd/cdi/CDIAnnotations.java

```diff
@@ -83,6 +84,19 @@ public boolean analyzeJar(Analyzer analyzer) throws Exception {
 			}
 		}
 
+		// If, after removing every instruction where 'discover=none', there are
+		// no items left we should just return.
+		if (header.entrySet()
+			.stream()
+			.map(e -> e.getValue()
+				.get("discover"))
+			.filter(Objects::nonNull)
+			.filter(discover -> Discover.valueOf(discover) == Discover.none)
```

Comment:
If the string is not a valid enum name, you will need to catch IllegalArgumentException from the `Discover.valueOf(String)` call.

For example:

https://github.com/bndtools/bnd/blob/abdf289ce7599d1595a29ec546774d56eb824f99/biz.aQute.bndlib/src/aQute/bnd/cdi/CDIAnnotations.java#L136-L139